> I cannot reproduce this error, but running valgrind I found another, possibly related one in Prima, which is fixed and committed now. Would you be able to take the latest Prima github snapshot and see if the error and coredump is still there?

I'd love to in principle, but as noted above, I can't reproduce this on my system, only on GitHub Actions CI. I did notice the above CI failed also on Windows, so I tried there (Strawberry 5.32 PDL Portable, with PDL 2.078 installed). I installed Prima fine, but P:D:P failed saying:

```
g++.exe Prima.def -o blib\arch\auto\PDL\Drawing\Prima\Prima.xs.dll -mdll -s -L"*\perl\lib\CORE" -L"*\c\lib" Prima.o   "*\perl\lib\CORE\libperl532.a" -lmoldname -lkernel32 -luser32 -lgdi32 -lwinspool -lcomdlg32 -ladvapi32 -lshell32 -lole32 -loleaut32 -lnetapi32 -luuid -lws2_32 -lmpr -lwinmm -lversion -lodbc32 -lodbccp32 -lcomctl32 -Wl,--enable-auto-image-base
ld.exe: Prima.o:Prima.c:(.text+0x63c): undefined reference to `gimme_the_mate'
[snip]
ld.exe: Prima.o:Prima.c:(.text+0x204d4): undefined reference to `prima_array_new'
ld.exe: Prima.o:Prima.c:(.text+0x20501): undefined reference to `prima_array_tie'
ld.exe: Prima.o:Prima.c:(.text+0x20cee): undefined reference to `prima_array_truncate'
[snip]
```

If the Prima changes since the last full release are minor, and include what you're saying is a real bug fix, you could make a new release and we can try this again?

--- a/lib/PDL/Demos/Prima.pm
+++ b/lib/PDL/Demos/Prima.pm
@@ -125,20 +125,15 @@ sub setup_gui {
 
 																	# Window
 	$gui->{window} = Prima::Window->create(
-		place => {
-			relx => 0.15, relwidth => 0.7, relheight => 0.7, rely => 0.15,
+		place     => {
+			relx   => 0.15, relwidth => 0.7, relheight => 0.7, rely => 0.15,
 			anchor => 'sw',
 		},
-		sizeMax => [600, 800],
-		sizeMin => [600, 800],
-		text => 'PDL::Graphics::Prima Demo',
-		onDestroy => sub {
-			require Prima::Utils;
-			# Throw an exception after destruction is complete so that we
-			# break out of the $::application->go loop.
-			Prima::Utils::post(sub { die 'time to exit the event loop' });
-		},
-		onKeyUp => \&keypress_handler,
+		sizeMax   => [600, 800],
+		sizeMin   => [600, 800],
+		text      => 'PDL::Graphics::Prima Demo',
+		onDestroy => sub { $::application->stop },
+		onKeyUp   => \&keypress_handler,
 	);
 	$gui->{window}->font->size(12);
 																		# Title

